recipes/xnnpack/all/conanfile.py

```diff
@@ -0,0 +1,101 @@
+from conans import ConanFile, CMake, tools
+from conans.errors import ConanInvalidConfiguration
+import glob
+import os
+
+
+class XnnpackConan(ConanFile):
+    name = "xnnpack"
+    description = "XNNPACK is a highly optimized library of floating-point " \
+                  "neural network inference operators for ARM, WebAssembly, " \
+                  "and x86 platforms."
+    license = "BSD-3-Clause"
+    topics = ("conan", "xnnpack", "neural-network", "inference", "multithreading",
+              "inference-optimization", "matrix-multiplication", "simd")
+    homepage = "https://github.com/google/XNNPACK"
+    url = "https://github.com/conan-io/conan-center-index"
+
+    settings = "os", "arch", "compiler", "build_type"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+        "assembly": [True, False],
+        "memopt": [True, False],
+        "sparse": [True, False],
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+        "assembly": True,
+        "memopt": True,
+        "sparse": True,
+    }
+
+    exports_sources = "CMakeLists.txt"
+    generators = "cmake"
+    _cmake = None
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.shared:
+            del self.options.fPIC
+        del self.settings.compiler.libcxx
+        del self.settings.compiler.cppstd
```

Comment:
```suggestion
        del self.settings.compiler.cppstd

    def validate(self):
```

it can be moved to validate.